fix: workaround customized local envs

swatDong opened the pull request:
Load `.env.teamsfx.local` when local debugging. This is part of custom local debug feature but needed by multi-env, so include this in go production release.

@dooriya please help review.
@KennethBWSong FYI, this is for customized local envs in samples.

swatDong commented:
E2E TEST: https://github.com/OfficeDev/TeamsFx-UI-Test/blob/b10d19213424ab2aece2bd286521fc07cbca36f9/src/ui-test/localdebug/localdebug-tabbot.test.ts#L7

a1exwang commented on packages/vscode-extension/src/debug/commonUtils.ts:
```diff
@@ -38,6 +38,24 @@ export async function getLocalEnv(): Promise<{ [key: string]: string } | undefin
     // use localSettings.json as input to generate the local debug envs
     env = await getLocalDebugEnvs();
+
+    // try load .env.teamsfx.local for tab/func/bot
```
does CLI need this fix?

swatDong replied:
Good catch. I'll move this to local debug plugin.

swatDong replied:
Moved changes to local debug plugin so it works for CLI as well. Added test case.

dooriya commented on packages/vscode-extension/src/debug/commonUtils.ts:
```diff
@@ -38,6 +38,24 @@ export async function getLocalEnv(): Promise<{ [key: string]: string } | undefin
+
+    // try load .env.teamsfx.local for tab/func/bot
+    // TODO: remove this workaround after fully supporting custom local debug
+    env = await appendEnvWithPrefix(
```
How about the CLI support for customized environment variables?

swatDong replied:
Good catch. I'll move this to local debug plugin.

swatDong replied:
Moved changes to local debug plugin so it works for CLI as well. Added test case.

kuojianlu commented:
How to let user know this customization? Is it necessary to update [the help doc](https://github.com/OfficeDev/TeamsFx/blob/main/docs/fx-core/localdebug-help.md)?

swatDong commented:
> How to let user know this customization? Is it necessary to update [the help doc](https://github.com/OfficeDev/TeamsFx/blob/main/docs/fx-core/localdebug-help.md)?

Updated help doc. Added section about environment variable customization.
